### ISSUE TITLE
Fix to demo script only working as root?

### ISSUE THREAD

**Issue by jwhendy** [J-Pai/EnvTrackerNode#1]
Greetings,

I've been unable to read anything other than the accelerometer and control the LED matrix on my sense hat for a long time. I recently re-visited an [old issue](https://github.com/astro-pi/python-sense-hat/issues/79), and someone referred to the blacklist instructions here. I followed them, things still didn't work, but then I was able to succeed by installing RTIMULib and sense_hat to the system and running as root.

You allude to this as a possible problem in the readme, but now how to fix it. I have confirmed this, which was the only obvious permissions related reason:
```
$ cat /etc/udev/rules.d/99-i2c.rules 
KERNEL=="i2c-[0-7]",MODE="0666"
```

Thanks for any input. Apologies for asking here, as you're not directly responsible... but there's no real solution at AstroPi; everyone points to "this is something wrong with RTIMULib" but most of their repos don't even have an issues tab, and I [posted](https://github.com/jeff-loughlin/RTIMULib2/issues/2) in one that does and have not received a response.

**Comment by J-Pai** [comment]
Are you able to read data off of the HAT using sudo? 

I'm assuming your environment is setup similarly to mine (which is a RPi running Ubuntu).

**Comment by J-Pai** [comment]
Ok, looking at your original comments, it looks like it works with sudo/root, but you want it to run the test script without requiring root. On my RPi, I do not have this issue and I am able to execute the test script (sense_hat_demo.py) without sudo.

With the MODE set to 0666, this should have allowed any user (and subsequently any application associated with user) to interact with the i2c device. 

```
KERNEL=="i2c-[0-7]",MODE="0666"
```

I know this seems pretty stupid, but sometimes udevadm does not fully refresh the rules after you add new *.rules file. Have you tried rebooting your device?
If this still doesn't work, you can potentially try and force the i2c devices to associate with a specific group.
Since you are using arch, I'm not entirely sure what default groups are associated with your user, but you can add `GROUP="dialout"` (or whatever group your user is currently assigned to) to the rules file (and reboot your raspberry pi).

e.g.
```
KERNEL=="i2c-[0-7]",MODE="0666",GROUP="dialout"
```

Sorry if I'm not too helpful here, I'm not too sure about the differences between Arch and Ubuntu and the way both distros set up their default permissions.

**Comment by jwhendy** [comment]
@J-Pai ok, that was helpful with respect to something to hunt for. I don't know why arch vs. ubuntu are different in this respect (namely, why `MODE` doesn't seem to work), but I found [this post](https://lexruee.ch/setting-i2c-permissions-for-non-root-users.html), which did the trick.

It suggests changing ownership of `/dev/i2c-1` to a newly created `i2c` group, and then adding one''s user to that group. I did not adjust `/etc/udev/rules.d` files and it worked. This [rpi SE post](https://raspberrypi.stackexchange.com/questions/4468/i2c-group-on-arch) had a similar suggestion (second answer far below as of today).

Thanks, and I'll consider this resolved!
